modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/list/List.tsx

```diff
@@ -3,12 +3,148 @@
+import {Body, Cell, Head, Row, Table} from '@clayui/core';
+import {ClayInput} from '@clayui/form';
+import ClayLayout from '@clayui/layout';
+import classNames from 'classnames';
+import {openModal} from 'frontend-js-web';
 import React from 'react';
 
+import {IFDSViewSectionProps} from '../../../FDSView';
 import {IBaseVisualizationMode} from '../VisualizationModes';
+import {IFDSField} from '../table/Table';
+import AddFieldsModalContent from '../table/modal_content/AddFieldsModalContent';
 
 export interface IList extends IBaseVisualizationMode<'list'> {}
 
-export default function List() {
-	return <>List</>;
+interface IListField {
+	fieldId: string;
+	label: string;
+}
+
+const LIST_VISUALIZATION_MODE_FIELDS: IListField[] = [
+	{fieldId: 'title', label: Liferay.Language.get('title')},
+	{fieldId: 'description', label: Liferay.Language.get('description')},
+	{fieldId: 'symbol', label: Liferay.Language.get('symbol')},
+	{fieldId: 'link', label: Liferay.Language.get('link')},
+	{fieldId: 'label', label: Liferay.Language.get('label')},
+];
+
+const LABEL_CELL_WIDTH = '30%';
+
+export default function List(props: IFDSViewSectionProps) {
+	const [fieldValues, setFieldValues] = React.useState<
+		Record<IListField['fieldId'], IFDSField>
+	>({});
+
+	const onValueChange = (
+		fieldId: IListField['fieldId'],
+		value: IFDSField
+	) => {
+		setFieldValues({
+			...fieldValues,
+			[fieldId]: value,
+		});
+	};
+
+	return (
+		<ClayLayout.ContentCol className="c-gap-4">
+			<ClayAlert
+				displayType="info"
+				title={`${Liferay.Language.get('info')}:`}
+				variant="stripe"
+			>
+				{Liferay.Language.get(
+					'list-visualization-mode-wont-be-shown-until-you-assign-at-least-one-field-to-a-list-element'
+				)}
+			</ClayAlert>
+
+			<Table className="mb-0" columnsVisibility={false}>
+				<Head>
+					<Cell width={LABEL_CELL_WIDTH}>
+						{Liferay.Language.get('list-element')}
+					</Cell>
+
+					<Cell className="border-left-0">
+						{Liferay.Language.get('field')}
+					</Cell>
+				</Head>
+
+				<Body>
+					{LIST_VISUALIZATION_MODE_FIELDS.map((field) => (
+						<ListField
+							field={field}
+							key={field.fieldId}
+							modalProps={props}
+							onValueChange={(fdsField) =>
+								onValueChange(field.fieldId, fdsField)
+							}
+							value={fieldValues[field.fieldId]}
+						/>
+					))}
+				</Body>
+			</Table>
+		</ClayLayout.ContentCol>
+	);
+}
+
+interface IListFieldProps {
+	field: IListField;
+	modalProps: IFDSViewSectionProps;
+	onValueChange: (fdsField: IFDSField) => void;
+	value?: IFDSField;
+}
+
+function ListField({field, modalProps, onValueChange, value}: IListFieldProps) {
+	const onClick = () => {
+		openModal({
+			contentComponent: ({closeModal}: {closeModal: Function}) => (
+				<AddFieldsModalContent
+					{...modalProps}
+					closeModal={closeModal}
+					onSave={({createdFDSFields: [createdFDSField]}) =>
+						onValueChange(createdFDSField)
+					}
```

Comment:
Would it be cleaner to call this prop `onSave`? Then, in parent, it would be
```js
<ListField
    onSave={(fdsField) =>
       // update top state
    }
/>
```

Reply:
I'm not sure about that, technically then this function is called that value hasn't been saved, just updated in the field component...

Reply:
Let me clarify what I mean:
```js
onSave={({createdFDSFields: [createdFDSField]}) =>
   onValueChange(createdFDSField)
}
```
This is a callback that happens on save. This is also the only place where `onValueChange` is used in the component. If I understand your intent correctly, you are pushing down `onValueChange` to the child component, as a prop. What I'm suggesting is a different way of thinking: we can expose `onSave` callback in the parent, and consequently keep `onValueChange` isolated in the parent.

Reply:
Alright, got it :smile: